src/main/java/bruno/TaskList.java

```diff
@@ -0,0 +1,255 @@
+        }
+        if (Integer.parseInt(markVal) > tasks.size()) {
+            throw new BrunoIndexOutOfBoundsException("mark");
+        }
+        if (Integer.parseInt(markVal) < 0) {
+            throw new BrunoNegativeArgException("mark");
+        }
+        tasks.get(Integer.parseInt(markVal) - 1).markAsDone();
+        storage.writeToFile(this);
+        String taskInfo =
+                "Woof Woof! I have marked the task as done.\n" + tasks.get(Integer.parseInt(markVal) - 1)
+                        .getString();
+        return taskInfo;
+    }
+
+    /**
+     * Unmarks a task to show that it is not done, from the list of tasks.
+     *
+     * @param task The Unmark command input by the user.
+     * @throws BrunoException Thrown if user tries to unmark an invalid task.
+     */
+    public String unmarkTask(String task) throws BrunoException {
+        String unmarkVal = task.split(" ")[1];
+        try {
+            int a = Integer.parseInt(unmarkVal);
+        } catch (NumberFormatException e) {
+            throw new BrunoIntegerMismatchException("mark");
+        }
+        if (Integer.parseInt(unmarkVal) > tasks.size()) {
+            throw new BrunoIndexOutOfBoundsException("unmark");
+        }
+        if (Integer.parseInt(unmarkVal) < 0) {
+            throw new BrunoNegativeArgException("unmark");
+        }
+        tasks.get(Integer.parseInt(unmarkVal) - 1).unMark();
+        storage.writeToFile(this);
+        String taskInfo = "OK, I have marked the task as not done yet.\n" + tasks.get(
+                Integer.parseInt(unmarkVal) - 1).getString();
+        return taskInfo;
+    }
+
+    /**
+     * Deletes a task from the list of tasks.
+     *
+     * @param task The Delete command input by the user.
+     * @throws BrunoException Thrown if user tries to delete an invalid task.
+     */
+    public String deleteTask(String task) throws BrunoException {
+        String deleteVal = task.split(" ")[1];
+        try {
+            int a = Integer.parseInt(deleteVal);
+        } catch (NumberFormatException e) {
+            throw new BrunoIntegerMismatchException("mark");
+        }
+        if (Integer.parseInt(deleteVal) > tasks.size()) {
+            throw new BrunoIndexOutOfBoundsException("delete");
+        }
+        if (Integer.parseInt(deleteVal) < 0) {
+            throw new BrunoNegativeArgException("delete");
+        }
+        String s1 = tasks.get(Integer.parseInt(deleteVal) - 1).getString();
+        tasks.remove(Integer.parseInt(deleteVal) - 1);
+        storage.writeToFile(this);
+        String taskInfo = "I have removed this task from your tasks:\n" + s1;
+        return taskInfo;
+    }
+
+    /**
+     * Displays the number of tasks in the tasks.
+     */
+    public String displayListSum() {
+        String taskInfo = "Now you have " + tasks.size() + (tasks.size() == 1 ? " task" : " tasks") + " "
+                + "in your "
+                + "tasks.";
```

Comment:
Do we need the indentation to be like this? Maybe the indentation came when you accidentally edited this line?
```suggestion
                String taskInfo = "Now you have " + tasks.size() 
                            + (tasks.size() == 1 ? " task" : " tasks") + " in your tasks.";
```